Tests/Packet++Test/Tests/IcmpV6Tests.cpp

```diff
@@ -0,0 +1,287 @@
+#include "../TestDefinition.h"
+#include "../Utils/TestUtils.h"
+#include "EndianPortable.h"
+#include "EthLayer.h"
+#include "IPv6Layer.h"
+#include "IcmpV6Layer.h"
+#include "Logger.h"
+#include "MacAddress.h"
+#include "NdpLayer.h"
+#include "Packet.h"
+#include "SystemUtils.h"
+#include <sstream>
+
+PTF_TEST_CASE(IcmpV6ParsingTest)
```

Comment:
Can we find another ICMPv6 packet that is not a request, reply, neighbor solicitation or neighbor advertisement and try to parse it too?

Reply:
currently i don't have time to do this, generating new/other ipv6 frames can sometimes be very time consuming, maybe someone else could do this or maybe we can bring this PR to dev branch and adding new frames later on?

Reply:
Don't worry about it, I'll open another PR right after we merge this one and add this